Mods/vcmi/config/vcmi/towerFactions.json

```diff
@@ -0,0 +1,101 @@
+{
+	"core:castle" :
+	{
+		"town" :
+		{
+			"siege" :
+			{
+				"towerIconSmall" : "vcmi/creatureIcons/towerSmall",
+				"towerIconLarge" : "vcmi/creatureIcons/towerLarge",
+			}
+		}
+	},
```

Comment:
Long-term solution is probably merge our virtual "core" mod and helper "vcmi" mod into single non-virtual mod and move all moddable config files into such mod. (probably name it "core" for compatibility with existing mods).
This would keep all data in a single place & less special cases to handle in our code.

As for overriding - for new factions nothing would change. They don't use this json in any way. Compatibility icons for unupdated mods are set in arrow tower "creature", which is shared across all towns.

vcmi-extras - will recheck tomorrow morning, worst-case scenario is that that mod will need to be marked as depended on "vcmi" mod, to guarantee correct load order.